src/app/space/create/deployments/services/deployments.service.ts

```diff
@@ -129,6 +129,18 @@ export class DeploymentsService implements OnDestroy {
       .distinctUntilChanged();
   }
 
+  isApplicationsDeployed(spaceId: string, environments: string[]): Observable<boolean> {
+    let bool = false;
+    for (let i = 0; i < environments.length; i++) {
+      this.isDeployedInEnvironment(spaceId, environments[i]).subscribe(b => {
+        if (b) {
+          bool = true;
+        }
+      });
+    }
+    return Observable.of(bool);
+  }
```

Comment:
This doesn't look like it's going to work properly. `isDeployedInEnvironment` is async, but you're looping and returning synchronously. You should be using RxJS streaming operators for this logic, something like:

```
Observable.of(environments)
  .map((environment: string): [string, string] => [spaceId, environment])
  .flatMap((pair: [string, string]): Observable<boolean> => this.isDeployedInEnvironment(pair[0], pair[1]))
  .every((deployed: boolean): boolean => deployed);
```
```

Reply:
```
Observable.combineLatest(
  environments.map(environment => this.isDeployedInEnvironment(spaceId, environment))
)
  .map((deployed: boolean[]): boolean => deployed.some(v => v))
```

is probably more like it.

Reply:
Good call, that's a nice solution thanks - I've gone forward with your function in this latest revision.